### PR TITLE
feat: use hook to define current year for licenses and copyright headers

### PR DESCRIPTION
**Pull request by renancaraujo**
## Description

Use mason hooks to set the copyright years to be generated upon brick usage, not creation. 

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: brick/__brick__/lib/counter/counter.dart [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2022, Very Good Ventures
+// Copyright (c) {{current_year}}, Very Good Ventures
```
Comment (felangel) [inline brick/__brick__/lib/counter/counter.dart:1]: We shouldn't need to modify the brick files manually. They should be updated by the bot when the PR is merged

File: brick/hooks/pre_gen.dart [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import 'dart:io';
```
[the remark this answers is not in the document]
Comment (renancaraujo) [reply brick/hooks/pre_gen.dart]: No

File: brick/__brick__/lib/counter/counter.dart (hunk 1 above)
Comment (renancaraujo) [reply brick/__brick__/lib/counter/counter.dart]: Reverted the changes.

File: brick/hooks/pre_gen.dart [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import 'package:mason/mason.dart';
+
+void run(HookContext context) {
+  final year = DateTime.now().year;
+  context.vars['current_year'] = year;
```
[the remark this answers is not in the document]
Comment (renancaraujo) [reply brick/hooks/pre_gen.dart]: lol, just pushed that change

**Pull request merged**